### DISCUSSION TITLE
Rand3DElasticd does not work as expected

### DISCUSSION THREAD

**Discussion by Spenhouet** [Project-MONAI/MONAI#3320] [Q&A]
**Describe the bug**

Given the following image and label:
![image](https://user-images.githubusercontent.com/7819068/141314133-b37f21c1-48eb-413c-bba4-21c4e774f745.png)

When applying `Rand3DElasticd`, the image and labels no longer match:

![image](https://user-images.githubusercontent.com/7819068/141323150-a7158815-719b-4d5c-b4b1-89efb5e22c8b.png)

**To Reproduce**

1. Load these files: [example_files.zip](https://github.com/Project-MONAI/MONAI/files/7521159/example_files.zip)
2. Execute this pipeline:
```python
from pathlib import Path

import numpy as np
from monai.transforms.compose import Compose
from monai.transforms.io.dictionary import LoadImaged, SaveImaged
from monai.transforms.spatial.dictionary import Rand3DElasticd
from monai.transforms.utility.dictionary import (EnsureChannelFirstd, ToTensord)
from monai.utils.enums import GridSampleMode

path = Path('scratch/rand_affine')

T1_KEY = 't1'
LABEL_KEY = 'labels'
FILE_KEYS = [T1_KEY, LABEL_KEY]
SAMPLE_MODES = [GridSampleMode.BILINEAR, GridSampleMode.NEAREST]

data = {
    T1_KEY: path / 't1.nii.gz',
    LABEL_KEY: path / 'labels.nii.gz',
}

process = Compose([
    LoadImaged(FILE_KEYS),
    ToTensord(FILE_KEYS),
    EnsureChannelFirstd(FILE_KEYS),
    Rand3DElasticd(
        FILE_KEYS,
        sigma_range=(5, 7),
        magnitude_range=(50, 150),
        mode=SAMPLE_MODES,
        prob=1.0,
    ),
    SaveImaged(
        FILE_KEYS,
        output_dir=path,
        output_postfix=f"out",
        resample=False,
        output_dtype=np.float32,
        separate_folder=False,
        print_log=False,
    )
])

process([data])
```

**Expected behavior**
The same random elastic deformation is applied to all images.
After the deformation the labels still match the image.

**Additional context**
I did not yet identify the issue.

**Comment by wyli** [comment]
probably duplicate of https://github.com/Project-MONAI/MONAI/discussions/3319 I'm converting it to discussions

**Comment by Spenhouet** [comment]
@Nic-Ma @ericspod @wyli @rijobro This was closed due to an unrelated issue with itk-snap. I don't know why this was the reason for closing this bug report? Could you clarify and reopen the issue?
